inc/zoo/swar/associative_iteration.h

```diff
@@ -106,6 +106,36 @@ fullAddition(SWAR<NB, B> s1, SWAR<NB, B> s2) {
     return { result, BS{carry.value()}, BS{overflow.value()} };
 };
 
+static_assert( SWAR<4, u32>(0x1111'1111).value() == fullAddition(SWAR<4, u32>(0x0111'1101), SWAR<4, u32>(0x1000'0010)).result.value());
+static_assert( SWAR<4, u32>(0x0000'0000).value() == fullAddition(SWAR<4, u32>(0x0111'1101), SWAR<4, u32>(0x1000'0010)).carry.value());
+static_assert( SWAR<4, u32>(0x0000'0000).value() == fullAddition(SWAR<4, u32>(0x0111'1101), SWAR<4, u32>(0x1000'0010)).overflow.value());
+
+template<int NB, typename B>
+constexpr ArithmeticResultTriplet<NB, B>
+fullSaturatedAddition(SWAR<NB, B> s1, SWAR<NB, B> s2) {

```

Comment:
I think there are two flavors of saturation: unsigned and signed.

Reply:
What about clamping?

Reply:
Name fixed, clamping is for a future PR.